client/http-client/src/client.rs

```diff
@@ -377,34 +378,15 @@ where
 		let id = self.id_manager.next_request_id();
 		let params = params.to_rpc_params()?;
 
-		let request = RequestSer::borrowed(&id, &method, params.as_deref());
-		let raw = serde_json::to_string(&request).map_err(Error::ParseError)?;
-
-		let fut = self.transport.send_and_read_body(raw);
-		let body = match tokio::time::timeout(self.request_timeout, fut).await {
-			Ok(Ok(body)) => body,
-			Err(_e) => {
-				return Err(Error::RequestTimeout);
-			}
-			Ok(Err(e)) => {
-				return Err(Error::Transport(e.into()));
-			}
-		};
-
-		// NOTE: it's decoded first to `JsonRawValue` and then to `R` below to get
-		// a better error message if `R` couldn't be decoded.
-		let response = ResponseSuccess::try_from(serde_json::from_slice::<Response<&JsonRawValue>>(&body)?)?;
+		let request = Request::borrowed(method, params.as_deref(), id.clone());
+		let rp =
```

Comment:
/cc @jsdw this into_method_call() is the annoyance I was talking about

Reply:
Aah gotcha! Yeah the only "nice" fix would be if we can use `impl Future<Output = Self::CallResponse>` etc in the trait and then have a different type for method, call and batch response or whatever.

This is all good for now though; hard to avoid without overcomplicating things unless `impl Trait` works ok!